test/libgit2.jl

```diff
@@ -41,206 +70,290 @@ begin
     end
 end
 
-function credentials!(cfg::LibGit2.GitConfig, usr="Test User", usr_email="Test@User.com")
-    git_user = LibGit2.get(cfg, "user.name", usr)
-    usr==git_user && LibGit2.set!(cfg, "user.name", usr)
-    git_user_email = LibGit2.get(cfg, "user.email", usr_email)
-    usr_email==git_user_email && LibGit2.set!(cfg, "user.email", usr_email)
+@testset "Signature" begin
+    sig = LibGit2.Signature("AAA", "AAA@BBB.COM", round(time(), 0), 0)
+    git_sig = convert(LibGit2.GitSignature, sig)
+    sig2 = LibGit2.Signature(git_sig)
+    finalize(git_sig)
+    @test sig.name == sig2.name
+    @test sig.email == sig2.email
+    @test sig.time == sig2.time
 end
 
-#TODO: tests need 'user.name' & 'user.email' in config ???
-LibGit2.with(LibGit2.GitConfig) do cfg
-    credentials!(cfg)
-end
+temp_dir() do dir
 
-function temp_dir(fn::Function, remove_tmp_dir::Bool=true)
-    tmpdir = joinpath(tempdir(),randstring())
-    @test !isdir(tmpdir)
-    try
-        mkdir(tmpdir)
-        @test isdir(tmpdir)
-        fn(tmpdir)
-    finally
-        remove_tmp_dir && rm(tmpdir, recursive=true)
+    # test parameters
+    repo_url = "github.com/JuliaLang/Example.jl"
```

Comment:
cloning this requires net access, right? add libgit2-online to the travis and appveyor yml's